src/basicfuns.jl

```diff
@@ -196,16 +201,20 @@ end
 
 Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and handling non-finite values.
 """
-function logaddexp(x::T, y::T) where T<:Real
-    # x or y is  NaN  =>  NaN
-    # x or y is +Inf  => +Inf
-    # x or y is -Inf  => other value
-    isfinite(x) && isfinite(y) || return max(x,y)
-    x > y ? x + log1p(exp(y - x)) : y + log1p(exp(x - y))
+function logaddexp(x::Real, y::Real)
+    # ensure Δ = 0 if x = y = Inf
+    Δ = ifelse(x == y, zero(x - y), abs(x - y))
+    max(x, y) + log1pexp(-Δ)
 end
-logaddexp(x::Real, y::Real) = logaddexp(promote(x, y)...)
 
-Base.@deprecate logsumexp(x::Real, y::Real) logaddexp(x,y)
+Base.@deprecate logsumexp(x::Real, y::Real) logaddexp(x, y)
+
+"""
+    logsubexp(x, y)
+
+Return `log(abs(e^x - e^y))`, preserving numerical accuracy.
+"""
+logsubexp(x::Real, y::Real) = max(x, y) + log1mexp(-abs(x - y))
```

Comment:
Is there no need to handle Inf as in `logaddexp`?

Reply:
No. Here it is okay that `logsubexp(Inf,Inf)` returns `NaN`. I test for that explicitly.